PWGEM/PhotonMeson/Tasks/Pi0EtaToGammaGammaMC.cxx

```diff
@@ -0,0 +1,346 @@
+    }
+  }
+
+  template <typename TMCParticle1, typename TMCParticle2, typename TMCParticles>
+  int FindCommonMotherFrom2Prongs(TMCParticle1 const& p1, TMCParticle2 const& p2, const int expected_pdg1, const int expected_pdg2, const int expected_mother_pdg, TMCParticles const& mcparticles)
+  {
+    if (p1.globalIndex() == p2.globalIndex())
+      return -1; // mc particle p1 and p2 is identical. reject.
+
+    if (p1.pdgCode() != expected_pdg1)
+      return -1;
+    if (p2.pdgCode() != expected_pdg2)
+      return -1;
+
+    if (!p1.has_mothers())
+      return -1;
+    if (!p2.has_mothers())
+      return -1;
+
+    // LOGF(info,"original motherid1 = %d , motherid2 = %d", p1.mothersIds()[0], p2.mothersIds()[0]);
+
+    int motherid1 = p1.mothersIds()[0];
+    auto mother1 = mcparticles.iteratorAt(motherid1);
+    int mother1_pdg = mother1.pdgCode();
+
+    int motherid2 = p2.mothersIds()[0];
+    auto mother2 = mcparticles.iteratorAt(motherid2);
+    int mother2_pdg = mother2.pdgCode();
+
+    // LOGF(info,"motherid1 = %d , motherid2 = %d", motherid1, motherid2);
+
+    if (motherid1 != motherid2)
+      return -1;
+    if (mother1_pdg != mother2_pdg)
+      return -1;
+    if (mother1_pdg != expected_mother_pdg)
+      return -1;
+    return motherid1;
+  }
+
+  Preslice<MyV0Photons> perCollision_pcm = aod::v0photon::collisionId;
+
+  template <PairType pairtype, typename TEvents, typename TPhotons1, typename TPhotons2, typename TPreslice1, typename TPreslice2, typename TV0Legs, typename TMCParticles>
+  void TruePairing(TEvents const& collisions, TPhotons1 const& photons1, TPhotons2 const& photons2, TPreslice1 const& perCollision1, TPreslice2 const& perCollision2, TV0Legs const& v0legs, TMCParticles const& mcparticles)
+  {
+    constexpr int itmp = pairtype;
+    for (auto& collision : collisions) {
+      registry.fill(HIST(pairnames[itmp]) + HIST("/hCollisionCounter"), 1.0); // all
+      if (!collision.sel8()) {
+        continue;
+      }
+      registry.fill(HIST(pairnames[itmp]) + HIST("/hCollisionCounter"), 2.0); // FT0VX i.e. FT0and
+
+      if (collision.numContrib() < 0.5) {
+        continue;
+      }
+      registry.fill(HIST(pairnames[itmp]) + HIST("/hCollisionCounter"), 3.0); // Ncontrib > 0
+
+      if (abs(collision.posZ()) > 10.0) {
+        continue;
+      }
+      registry.fill(HIST(pairnames[itmp]) + HIST("/hCollisionCounter"), 4.0); //|Zvtx| < 10 cm
+
+      auto photons1_coll = photons1.sliceBy(perCollision1, collision.collisionId());
+      auto photons2_coll = photons2.sliceBy(perCollision2, collision.collisionId());
+
+      int pi0id = -1;
+      int etaid = -1;
+      if (pairtype == PairType::kPCMPCM || pairtype == PairType::kPHOSPHOS || pairtype == PairType::kEMCEMC) {
+        for (auto& [g1, g2] : combinations(CombinationsStrictlyUpperIndexPolicy(photons1_coll, photons2_coll))) {
```

Comment:
currently this is just implemented for PCM, correct? So for PHOs and EMCal the pi0id and etaid will stay -1?

Reply:
yes.

Reply:
Okay, thanks for your hard work here again :)